rpc/ethereum/namespaces/eth/filters/api.go

```diff
@@ -198,13 +210,22 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Su
 					continue
 				}
 
-				txHash := common.BytesToHash(tmtypes.Tx(data.Tx).Hash())
-
-				// To keep the original behavior, send a single tx hash in one notification.
-				// TODO(rjl493456442) Send a batch of tx hashes in one notification
-				err = notifier.Notify(rpcSub.ID, txHash)
+				tx, err := api.clientCtx.TxConfig.TxDecoder()(data.Tx)
 				if err != nil {
-					return
+					api.logger.Debug("fail to decode tx", "error", err.Error())
+					continue
+				}
+
+				for _, msg := range tx.GetMsgs() {
+					ethTx, ok := msg.(*evmtypes.MsgEthereumTx)
+					if ok {
+						// To keep the original behavior, send a single tx hash in one notification.
+						// TODO(rjl493456442) Send a batch of tx hashes in one notification
+						err = notifier.Notify(rpcSub.ID, common.HexToHash(ethTx.Hash))
+						if err != nil {
+							return
```

Comment:
log or return error?

Reply:
https://github.com/ethereum/go-ethereum/blob/master/eth/filters/api.go#L160
the error is ignored in go-ethereum, maybe it makes sense since it's run in a goroutine, I'll adopt that.